FirebaseCombineSwift/Sources/Firestore/CollectionReference+Combine.swift

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(Combine) && swift(>=5.0) && canImport(FirebaseFirestore)
+
+import Combine
+import FirebaseFirestore
+
+#if canImport(FirebaseFirestoreSwift)
+
+import FirebaseFirestoreSwift
+
+#endif
+
+@available(swift 5.0)
+@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+extension CollectionReference {
+    
+    // MARK: - Add Document
+    
+    /// Adds a new document to this collection with the specified data, assigning it a document ID
+    /// automatically.
+    /// - Parameter data: A `Dictionary` containing the data for the new document.
+    /// - Returns: A publisher emitting a `Void` value once the document has been successfully
+    /// written to the server. This publisher will not emits  while the client is offline, though
+    /// local changes will be visible immediately.
+    public func addDocument(data: [String: Any])
+    -> AnyPublisher<DocumentReference, Error> {
+        var reference: DocumentReference!
```

Comment:
The problem is that inside `.map{  }` the reference will be optional, despite this should never happen, otherwise the function is broken. At some of the chain the value should be unwrapped anyway.